### PR TITLE
Ported TtlPropertyPath in V3 .net SDK

### PR DESCRIPTION
**Pull request by simplynaveen20**
This change is ported from below master PR
https://msdata.visualstudio.com/_git/CosmosDB/pullrequest/203232?_a=overview

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosContainerSettings.cs [hunk 1]
```diff
@@ -303,6 +321,12 @@ public string PartitionKeyPath
         [JsonProperty(PropertyName = Constants.Properties.DefaultTimeToLive, NullValueHandling = NullValueHandling.Ignore)]
         internal int? InternalTimeToLive { get; set; }
 
+        /// <summary>
+        /// Internal property used as a helper to convert to the back-end type string?
+        /// </summary>
+        [JsonProperty(PropertyName = TimeToLivePropertyPathConstant, NullValueHandling = NullValueHandling.Ignore)]
+        internal string InternalTimeToLivePropertyPath { get; set; }
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosContainerSettings.cs]: Why not make it TimeToLivePropertyPath and public?

Public property is simple wrapper there is no logic/transformation.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs [hunk 2]
```diff
@@ -415,5 +415,54 @@ public async Task ImplicitConversion()
...
+        /// <summary>
+        /// This test only verifies that we are able to set the ttl property path correctly using SDK.
+        /// For ttl custom property path case, we don't do logical document filtering in backend and since expired
+        /// resource purger is not yet enabled in any fabric emulator; not verifying whether documents are getting ttl'ed
+        /// correctly or not. There are native tests to verify this functionality.
+        /// </summary>
+        [TestMethod]
+        public async Task TimeToLivePropertyPath()
+        {
+
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs]: Remove extra line

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs [hunk 3]
```diff
@@ -415,5 +415,54 @@ public async Task ImplicitConversion()
...
+            // verify removing the ttl property path
+            setting.TimeToLivePropertyPath = null;
+            containerResponse = await cosmosContainer.ReplaceAsync(setting);
+            cosmosContainer = containerResponse;
+            Assert.AreEqual(TimeSpan.FromSeconds(-1), containerResponse.Resource.DefaultTimeToLive);
+            Assert.IsNull(containerResponse.Resource.TimeToLivePropertyPath);
+
+            containerResponse = await cosmosContainer.DeleteAsync();
+            Assert.AreEqual(HttpStatusCode.NoContent, containerResponse.StatusCode);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs]: Also god to cover an item scenario as well.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosContainerSettings.cs (hunk 1 above)
Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosContainerSettings.cs]: Are we not avoiding  json contract from public property ?

Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosContainerSettings.cs]: Done

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs (hunk 3 above)
Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs]: Keeping this conversation open , as test case will fail with 5 min deletion cycle , will discuss it offline.

Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs]: As per conversation offline in scrum, removing reading expired item test

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosContainerSettings.cs [hunk 4]
```diff
@@ -90,6 +90,8 @@ namespace Microsoft.Azure.Cosmos
     /// <seealso cref="CosmosDatabaseSettings"/>
     public class CosmosContainerSettings
     {
+        private const string TimeToLivePropertyPathConstant = "ttlPropertyPath";
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosContainerSettings.cs]: Direct will be refreshed with my PR. Check if we can leverage that.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/SettingsContractTests.cs [hunk 5]
```diff
@@ -275,7 +275,7 @@ public void ContainerSettingsDefaults()
             string id = Guid.NewGuid().ToString();
             string pkPath = "/partitionKey";
 
-            SettingsContractTests.TypeAccessorGuard(typeof(CosmosContainerSettings), "Id", "UniqueKeyPolicy", "DefaultTimeToLive", "IndexingPolicy");
+            SettingsContractTests.TypeAccessorGuard(typeof(CosmosContainerSettings), "Id", "UniqueKeyPolicy", "DefaultTimeToLive", "IndexingPolicy", "TimeToLivePropertyPath");
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/SettingsContractTests.cs]: Thank u

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosContainerSettings.cs (hunk 4 above)
Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosContainerSettings.cs]: There are bunch of build error when we are increasing the direct to 3.0.0.25, do I need to wait for some PR (Work item)?

**Pull request merged**